cnxpublishing/db.py

```diff
@@ -1129,6 +1135,67 @@ def remove_acl(cursor, uuid_, permissions):
                        (uuid_, uid, permission,))
 
 
+def _upsert_users(cursor, user_ids, lookup_func):
+    """Upsert's user info into the database.
+    The model contains the user info as part of the role values.
+    """
+    user_ids = list(set(user_ids))  # cleanse data
+
+    # Check for existing records to update.
+    cursor.execute("SELECT username from users where username = ANY (%s)",
+                   (user_ids,))
+    try:
+        existing_user_ids = [x[0] for x in cursor.fetchall()]
+    except TypeError:
+        existing_user_ids = []
+    new_user_ids = [u for u in user_ids if u not in existing_user_ids]
+
+    # Update existing records.
+    for user_id in existing_user_ids:
+        # TODO only update based on a delta against the 'updated' column.
```

Comment:
doesn't really save anything: postgresql will need to form an entire new tuple anyway, and the data here is so small not sending the unchanged values isn't worth it. Simple is better. So, no TODO